crates/polars-ops/src/frame/join/merge_sorted.rs

```diff
@@ -159,6 +162,13 @@ fn series_to_merge_indicator(lhs: &Series, rhs: &Series) -> Vec<bool> {
             let rhs = rhs_s.binary().unwrap();
             get_merge_indicator(lhs.into_iter(), rhs.into_iter())
         },
+        #[cfg(feature = "dtype-struct")]
+        DataType::Struct(_) => {
+            let options = SortOptions::default();
+            let lhs = lhs_s.struct_().unwrap().get_row_encoded(options).unwrap();
```

Comment:
Ideally I would propagate the error given by the second unwrap here.

Reply:
Implemented.

What is your approach in this repo: should I mark the comments as resolved when I address them, or should I let the reviewer do it?

Reply:
I think as you resolve each comment you can mark them as resolved.

Reply:
Clarifying question: `get_row_encoded` always succeeds. In fact, none of the functions in `crates/polars-core/src/chunked_array/ops/row_encode.rs` can return an error, but the return type is still `PolarsResult`. Is this for a reason, e.g. making it easier to return errors in the future, or would it make sense to remove the `PolarsResult` from the return types of functions in `row_encode.rs` to simplify the code in the callers (such as `get_row_encoded`)?

Reply:
Ah, that might be a change that got missed over several refactors. I think for now leave it as is. I remember it was originally added to deal with things that cannot be row encoded, i.e. objects and weird categoricals. Those things might need to be readded at some point.